PAWS/Study Information/EnrollmentGroup.swift

```diff
@@ -0,0 +1,46 @@
+//
+// This source file is part of the PAWS application based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import FirebaseAuth
+import FirebaseFirestore
+import Spezi
+import SpeziAccount
+import SpeziFirebaseConfiguration
+import SwiftUI
+
+
+@Observable
+class EnrollmentGroup: Module, EnvironmentAccessible {
+    @ObservationIgnored @Dependency private var configureFirebaseApp: ConfigureFirebaseApp
+    var dateOfBirth: Date?
+    
+    var studyType: StudyType? {
+        guard let enrollmentDate = Auth.auth().currentUser?.metadata.creationDate,
+              let dateOfBirth,
+              let yearsOfAge = Calendar.current.dateComponents([.year], from: dateOfBirth, to: enrollmentDate).year else {
+            return nil
+        }
+        return yearsOfAge >= 18 ? .adult : .pediatric
+    }
+    
+    func configure() {
+        guard let uid = Auth.auth().currentUser?.uid else {
+            return
```

Comment:
In case that the user is not signed in we need to add a listener to the user state (https://firebase.google.com/docs/auth/ios/start) to ensure that we still get informed about the date of birth if the user is not signed in (first download of the app) and then signs in a first time.

It would generally be advised to refresh/re-register the listener at any authentication changes so I would wrap the complete code here in a separate function, call it on any changes of the user authentication and also call it at the very first launch of the module within the configure method.